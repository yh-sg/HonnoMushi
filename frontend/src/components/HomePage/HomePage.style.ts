import styled from "styled-components";

export const ContainerStyle = styled.div<{
	isBooksSearching?: boolean;
	count?: number;
	loading?: boolean;
}>`
	text-align: center;
	padding: ${(props): string => (props.isBooksSearching ? "3rem " : "")};
	margin-bottom: ${(props): string =>
		props.loading || props.count === 0 ? "31.5rem" : ""};
`;

export const LibraryImageStyle = styled.img`
	width: 75vw;
	border-radius: 2rem;
	margin: 2rem;
	box-shadow: 5px 5px 10px black;
`;

export const SearchBookStyle = styled.div`
	margin: 0 auto;
	& input {
		width: 16rem;
		padding: 0.3rem;
	}
`;
