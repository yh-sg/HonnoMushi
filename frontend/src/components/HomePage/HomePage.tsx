import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import library from "../../images/library.jpg";
import BooksSelection from "../../containers/BooksSelection/BooksSelection";
import { ContainerStyle, LibraryImageStyle } from "./HomePage.style";

const PageOne: React.FC = () => {
	return (
		<>
			<Router>
				<ContainerStyle>
					<LibraryImageStyle
						src={library}
						alt='classic vintage library of books'
					/>
					<h1>Search for a Book ðŸ“–</h1>
					<BooksSelection />
				</ContainerStyle>
			</Router>
		</>
	);
};

export default PageOne;
