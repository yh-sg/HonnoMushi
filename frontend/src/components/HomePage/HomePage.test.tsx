import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import HomePage from "./HomePage";

afterEach((): void => {
	cleanup();
});

describe("should render HomePage component", () => {
	test("should render HomePage component correctly", () => {
		// render a react component to the DOM
		const div = document.createElement("div");
		ReactDOM.render(<HomePage />, div);

		expect(div.querySelector("h3").textContent).toBe(
			"Find books by letters or alphabets"
		);
	});

	// render + cleanup will take snapshot
	test("should match snapshot", () => {
		const { asFragment } = render(<HomePage />);
		expect(asFragment()).toMatchSnapshot();
	});
});
