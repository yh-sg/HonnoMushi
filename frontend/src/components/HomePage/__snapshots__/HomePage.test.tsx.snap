// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT inYPAp"
  >
    <img
      alt="classic vintage library of books"
      class="sc-dlnjwi cTNnPW"
      src="library.jpg"
    />
    <h3>
      Find books by letters or alphabets
    </h3>
    <div
      class="sc-bdnxRM cZnSuQ"
    >
      <div
        class="row mt-3 mb-3"
      >
        <div>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            A
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            B
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            C
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            D
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            E
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            F
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            G
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            H
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            I
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            J
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            K
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            L
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            M
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            N
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            O
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            P
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            Q
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            R
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            S
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            T
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            U
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            V
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            W
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            X
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            Y
          </button>
          <button
            class="col-3 col-md-1 m-2 mb-3 btn btn-outline-success"
            data-testid="letterButton"
            type="button"
          >
            Z
          </button>
        </div>
      </div>
      <div
        class="row mb-3"
      >
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          0
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          1
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          2
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          3
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          4
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          5
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          6
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          7
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          8
        </button>
        <button
          class="col m-2 p-2 btn btn-outline-primary"
          data-testid="numberButton"
          type="button"
        >
          9
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
